lib/config/ChainConfigManager.ts

```diff
@@ -0,0 +1,300 @@
+    },
+    [ChainId.SEPOLIA]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: false,
+    },
+    [ChainId.CELO]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: true,
+    },
+    [ChainId.CELO_ALFAJORES]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: false,
+    },
+    [ChainId.BNB]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: true,
+    },
+    [ChainId.AVALANCHE]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: true,
+    },
+    [ChainId.BASE_GOERLI]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: false,
+    },
+    [ChainId.BASE]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: true,
+    },
+    [ChainId.BLAST]: {
+      routingTypes: [
+        {
+          routingType: RoutingType.CLASSIC,
+        },
+      ],
+      alarmEnabled: true,
+    },
+  };
+
+  private static _chainConfigsWithDependencies: { [chainId: number]: ChainConfig };
+  private static _reverseConfigs: { [routeType: string]: number[] };
+
+  /**
+   * Lazy load the full list
+   * Adds dependencies if they don't already exist
+   */
+  static get chainConfigsWithDependencies(): { [chainId: number]: ChainConfig } {
+    if (ChainConfigManager._chainConfigsWithDependencies) {
+      return ChainConfigManager._chainConfigsWithDependencies;
+    }
+    ChainConfigManager._chainConfigsWithDependencies = ChainConfigManager._chainConfigs;
+    for (const dependencyMapping in ChainConfigManager._routeDependencies) {
+      for (const dependency of ChainConfigManager._routeDependencies[dependencyMapping]) {
+        for (const chainId in ChainConfigManager._chainConfigs) {
+          let dependentPresent = false;
+          let dependencyPresent = false;
+          for (const supportedRoutingType of ChainConfigManager._chainConfigs[chainId].routingTypes) {
+            dependentPresent = dependentPresent || supportedRoutingType.routingType == dependencyMapping;
+            dependencyPresent = dependencyPresent || supportedRoutingType.routingType == dependency;
+          }
+          // If we have the dependent but not the dependency, add it
+          if (dependentPresent && !dependencyPresent) {
+            ChainConfigManager._chainConfigsWithDependencies[chainId].routingTypes.push({
+              routingType: dependency as RoutingType,
+            });
+          }
+        }
+      }
+    }
+    return ChainConfigManager._chainConfigsWithDependencies;
+  }
+
+  static get reverseConfigs(): { [routeType: string]: number[] } {
```

Comment:
nit: routeType => routingType for consistency

Reply:
also name - maybe `chainsByRoutingType` b/c "Reverse" isnt super clear what the "normal" direction is

Reply:
good feedback